pkg/extensions/predicate.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extensions
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+// ObjectPredicate returns true for 'create' and 'update' events. For updates, it only returns true when the extension
+// type has changed.
+func ObjectPredicate() predicate.Predicate {
```

Comment:
Why does the function description not mention the `delete` events when it actually handles them exactly like `create` events?

Reply:
Good question. I just moved the function, but I did not question it 😅